File: examples/python-example/src/ui/login_page.py
"""Login page UI component."""
from src.core.auth import Auth
from src.utils.logger import Logger


class Button:
    def __init__(self, label: str):
        self.label = label

    def render(self) -> str:
        return f"<button>{self.label}</button>"


class LoginPage:
    def __init__(self):
        self.auth = Auth()
        self.logger = Logger("LoginPage")

    def render(self) -> str:
        submit_btn = Button("Login")
        return f"<form>{submit_btn.render()}</form>"

    def handle_submit(self, username: str, password: str) -> None:
        success = self.auth.login(username, password)
        if success:
            self.logger.info("Login successful")
        else:
            self.logger.error("Login failed")

File: examples/typescript-example/src/ui/loginPage.ts
// UI login page
import { Auth } from '../core/auth';
import { Logger } from '../utils/logger';
import { Button } from './button';

export class LoginPage {
  private auth: Auth;
  private logger: Logger;

  constructor() {
    this.auth = new Auth();
    this.logger = new Logger('LoginPage');
  }

  render(): string {
    const submitBtn = new Button('Login');
    return `<form>${submitBtn.render()}</form>`;
  }

  handleSubmit(username: string, password: string): void {
    const success = this.auth.login(username, password);
    if (success) {
      this.logger.info('Login successful');
    } else {
      this.logger.error('Login failed');
    }
  }
}
